app/src/androidTest/java/com/github/se/eventradar/login/LoginTest.kt

```diff
@@ -0,0 +1,99 @@
+package com.github.se.eventradar.login
+
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.intent.Intents.intended
+import androidx.test.espresso.intent.matcher.IntentMatchers.toPackage
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.se.eventradar.screens.LoginScreen
+import com.github.se.eventradar.ui.MainActivity
+import com.kaspersky.kaspresso.testcases.api.testcase.TestCase
+import io.github.kakaocup.compose.node.element.ComposeScreen
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class LoginTest : TestCase() {
+  @get:Rule val composeTestRule = createAndroidComposeRule<MainActivity>()
+
+  @Test
+  fun titleAndButtonAreCorrectlyDisplayed() {
+    ComposeScreen.onComposeScreen<LoginScreen>(composeTestRule) {
+      // Test the UI elements
+      loginLogo { assertIsDisplayed() }
+      loginTitle {
+        assertIsDisplayed()
+        assertTextEquals("Event Radar")
+      }
+      loginButton {
+        assertIsDisplayed()
+        assertHasClickAction()
+      }
+    }
+  }
+
+  @Test
+  fun googleSignInReturnsValidActivityResult() {
+    Intents.init()
+    ComposeScreen.onComposeScreen<LoginScreen>(composeTestRule) {
+      loginButton {
+        assertIsDisplayed()
+        performClick()
+      }
+
+      // assert that an Intent resolving to Google Mobile Services has been sent (for sign-in)
+      intended(toPackage("com.google.android.gms"))
+    }
+    Intents.release()
+  }
+
+  // TODO: Add test for failed log-in
```

Comment:
We will have to discuss with the coaches if such a test is feasible.

Reply:
Yes, I think we should discuss it with the coaches, maybe the login could fail when there is no internet connection or when Firebase is down? 

The times when login failed for me are: when the firebase SHA1/google-services.json config file/or when the web-client id in the app was different.